keras_nlp/models/bert/bert_feature_extractor.py

```diff
@@ -0,0 +1,292 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""BERT feature extraction model."""
+
+import copy
+import os
+
+from tensorflow import keras
+
+from keras_nlp.models.bert.bert_backbone import BertBackbone
+from keras_nlp.models.bert.bert_preprocessor import BertPreprocessor
+from keras_nlp.models.bert.bert_presets import backbone_presets
+from keras_nlp.utils.pipeline_model import PipelineModel
+from keras_nlp.utils.python_utils import classproperty
+from keras_nlp.utils.python_utils import format_docstring
+
+PRESET_NAMES = ", ".join(list(backbone_presets))
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class BertFeatureExtractor(PipelineModel):
+    """An end-to-end BERT model for feature extraction task. This model
+    can't be compiled or fitted, and should be used as a submodel only.
+
+    For usage of this model with pre-trained weights, see
+    the `from_preset()` method.
+
+    This model can optionally be configured with a `preprocessor` layer, in
+    which case it will automatically apply preprocessing to raw inputs during
+    `fit()`, `predict()`, and `evaluate()`. This is done by default when
+    creating the model with `from_preset()`.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind.
+
+    Args:
+        backbone: A `keras_nlp.models.BertBackbone` instance.
+        dropout: float. The dropout probability value, applied after the dense
+            layer.
+        preprocessor: A `keras_nlp.models.BertPreprocessor` or `None`. If
+            `None`, this model will not apply preprocessing, and inputs should
+            be preprocessed before calling the model.
+
+    Examples:
+
+    ```python
+    # Call featurizer on the inputs.
+    preprocessed_features = {
+        "token_ids": tf.ones(shape=(2, 12), dtype=tf.int64),
+        "segment_ids": tf.constant(
+            [[0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0]] * 2, shape=(2, 12)
+        ),
+        "padding_mask": tf.constant(
+            [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]] * 2, shape=(2, 12)
+        ),
+    }
+
+    # Randomly initialize a BERT backbone.
+    backbone = keras_nlp.models.BertBackbone(
+        vocabulary_size=30552,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Create a BERT featurizer and evaluate on data
+    featurizer = keras_nlp.models.BertFeatureExtractor(
+        backbone,
+        preprocessor=None,
+    )
+
+    y_eval = featurizer(preprocessed)
+
+    # Access backbone programatically (e.g., to change `trainable`)
+    featurizer.backbone.trainable = False
+    
+    
+    # Using a classifier head    
+        
+    class CustomClassifier(keras.Model):
```

Comment:
Great, but in this case I'd recommend using a functional model since it's so much less verbose:
```python
outputs = backbone(outputs)["pooled_output"]
outputs = Dense(4)(outputs)
classifier = keras.Model(inputs, outputs)
```